fuels-signers/src/lib.rs

```diff
@@ -48,26 +46,26 @@ mod tests {
         let mut secret_seed = [0u8; 32];
         rng.fill_bytes(&mut secret_seed);
 
-        let secret =
-            SecretKey::from_slice(&secret_seed).expect("Failed to generate random secret!");
+        let secret = unsafe { SecretKey::from_bytes_unchecked(secret_seed) };
 
         let (provider, _) = setup_test_provider(vec![]).await;
         let wallet = LocalWallet::new_from_private_key(secret, provider).unwrap();
 
         let message = "my message";
 
-        let signature = wallet.sign_message(message.as_bytes()).await.unwrap();
+        let signature = wallet.sign_message(message).await.unwrap();
 
         // Check if signature is what we expect it to be
-        assert_eq!(signature.compact, Signature::from_str("0x8eeb238db1adea4152644f1cd827b552dfa9ab3f4939718bb45ca476d167c6512a656f4d4c7356bfb9561b14448c230c6e7e4bd781df5ee9e5999faa6495163d").unwrap().compact);
+        assert_eq!(signature, Signature::from_str("0x8eeb238db1adea4152644f1cd827b552dfa9ab3f4939718bb45ca476d167c6512a656f4d4c7356bfb9561b14448c230c6e7e4bd781df5ee9e5999faa6495163d").unwrap());
 
         // Recover address that signed the message
-        let recovered_address = signature.recover(message).unwrap();
+        let message = Message::new(message);
+        let recovered_address = signature.recover(&message).unwrap();
 
-        assert_eq!(wallet.address, recovered_address);
+        assert_eq!(wallet.address.as_ref(), recovered_address.hash().as_ref());
```

Comment:
why is it passed `as_ref`?

Reply:
```
mismatched types
expected struct `fuel_tx::Address`, found struct `fuel_tx::Bytes32`
```

I think they're both FuelVM's atomic types, with underlying arrays. At type-level, we can't know they're the same length, so we convert to `&[u8]` (an unbounded slice) with `as_ref` to compare.